java/calcite/src/main/java/com/mapd/parser/server/CalciteServerHandler.java

```diff
@@ -164,4 +171,72 @@ public void updateMetadata(String catalog, String table) throws TException {
       }
     }
   }
+
+  @Override
+  public List<TCompletionHint> getCompletionHints(String user, String session,
+      String catalog, String sql, int cursor) throws TException {
+    callCount++;
+    MapDParser parser;
+    try {
+      parser = (MapDParser) parserPool.borrowObject();
+    } catch (Exception ex) {
+      String msg = "Could not get Parse Item from pool: " + ex.getMessage();
+      MAPDLOGGER.error(msg);
+      throw new TException(msg);
+    }
+    MapDUser mapDUser = new MapDUser(user, session, catalog, mapdPort);
+    MAPDLOGGER.debug("getCompletionHints was called User: " + user + " Catalog: " + catalog + " sql: " + sql);
+    parser.setUser(mapDUser);
+
+    // remove last character if it is a ;
+    if (sql.charAt(sql.length() - 1) == ';') {
```

Comment:
we need a .trim sql prior to this, not that it makes sense to be calling for completion with a full sql stmt anyway.

It probably makes sense in this context to remove this code to remove `;` entirely

Reply:
I'm pretty sure neither is required, I remember testing with whitespace around the query. i'll remove it.